Modules/Tests/YosemiteTests/Tools/POS/POSCatalogSyncCoordinatorTests.swift

```diff
@@ -743,3 +582,230 @@ final class MockPOSCatalogFullSyncService: POSCatalogFullSyncServiceProtocol {
+        // Given - new user, catalog eligible
+        let coordinator = POSCatalogSyncCoordinator(
+            fullSyncService: mockSyncService,
+            incrementalSyncService: mockIncrementalSyncService,
+            grdbManager: grdbManager,
+            catalogEligibilityChecker: MockPOSLocalCatalogEligibilityService(), // Catalog eligible
+            siteSettings: mockSiteSettings
+        )
+        try createSiteInDatabase(siteID: sampleSiteID, lastFullSyncDate: nil)
+        mockSyncService.startFullSyncResult = .success(POSCatalog(products: [], variations: [], syncDate: .now))
+
+        // When
+        try await coordinator.performSmartSync(for: sampleSiteID)
+
+        // Then - sync should proceed
+        #expect(mockSyncService.startFullSyncCallCount == 1)
+    }
+
+    @Test func performSmartSync_records_first_sync_date_after_successful_sync() async throws {
+        // Given - new user
+        let coordinator = POSCatalogSyncCoordinator(
+            fullSyncService: mockSyncService,
+            incrementalSyncService: mockIncrementalSyncService,
+            grdbManager: grdbManager,
+            catalogEligibilityChecker: MockPOSLocalCatalogEligibilityService(),
+            siteSettings: mockSiteSettings
+        )
+        try createSiteInDatabase(siteID: sampleSiteID, lastFullSyncDate: nil)
+        mockSyncService.startFullSyncResult = .success(POSCatalog(products: [], variations: [], syncDate: .now))
+
+        // When
+        try await coordinator.performSmartSync(for: sampleSiteID)
+
+        // Then - first sync date should be recorded
+        #expect(mockSiteSettings.setFirstPOSCatalogSyncDateCalled == true)
+        #expect(mockSiteSettings.mockFirstPOSCatalogSyncDate != nil)
+    }
+
+    @Test func performSmartSync_does_not_overwrite_existing_first_sync_date() async throws {
+        // Given - existing user with first sync date
+        let originalDate = Date().addingTimeInterval(-10 * 24 * 60 * 60) // 10 days ago
+        mockSiteSettings.mockFirstPOSCatalogSyncDate = originalDate
+
+        let coordinator = POSCatalogSyncCoordinator(
+            fullSyncService: mockSyncService,
+            incrementalSyncService: mockIncrementalSyncService,
+            grdbManager: grdbManager,
+            catalogEligibilityChecker: MockPOSLocalCatalogEligibilityService(),
+            siteSettings: mockSiteSettings
+        )
+        try createSiteInDatabase(siteID: sampleSiteID, lastFullSyncDate: nil)
+        mockSyncService.startFullSyncResult = .success(POSCatalog(products: [], variations: [], syncDate: .now))
+
+        // When
+        try await coordinator.performSmartSync(for: sampleSiteID)
+
+        // Then - first sync date should remain unchanged
+        #expect(mockSiteSettings.mockFirstPOSCatalogSyncDate == originalDate)
+    }
+
+    @Test func performSmartSync_proceeds_for_new_user_within_30_day_grace_period() async throws {
```

Comment:
I wasn't quite sure what the test was doing when first checked, from what I understand we want to check if incremental sync is performed here, maybe we can rename this one a bit to make it easier?

```suggestion
    @Test func performSmartSync_proceeds_with_incremental_sync_for_new_user_within_30_day_grace_period() async throws {
```

or 
```suggestion
    @Test func  performSmartSync_when_first_sync_within_grace_period_then_triggers_incremental_sync() async throws {
```